homeassistant/helpers/service_info/mqtt.py

```diff
@@ -0,0 +1,20 @@
+"""MQTT Discovery data."""
+from dataclasses import dataclass
+import datetime as dt
+from typing import Union
+
+from homeassistant.data_entry_flow import BaseServiceInfo
+
+ReceivePayloadType = Union[str, bytes]
+
+
+@dataclass
+class MqttServiceInfo(BaseServiceInfo):
+    """Prepared info from mqtt entries."""
+
+    topic: str
+    payload: ReceivePayloadType
+    qos: int
+    retain: bool
+    subscribed_topic: str
+    timestamp: dt.datetime
```

Comment:
Instead of this, can we move it to `homeassistant/components/mqtt/models.py`?

Reply:
> Instead of this, can we move it to `homeassistant/components/mqtt/models.py`?

I tried something similar first but it didn't solve the issue for the reason @epenet pointed out.

Reply:
Why is `homeassistant/components/mqtt/__init__.py` loaded if `homeassistant/components/mqtt/models.py` is imported?

Reply:
Imports always have to figure out how to load everything below a path, as part of that python always imports `__init__.py` first if it exists.

https://docs.python.org/3.10/tutorial/modules.html#packages

Reply:
OK. What's the actual problem with `mqtt/__init__.py`? I thought it was related to importing all MQTT platforms, but it doesn't do that anymore.

Reply:
There nothing wrong with the code, other then its being loaded unexpectedly or at the wrong time

If they don't have MQTT they get the whole integration read from disk, compiled, and loaded into memory even if they will never use it

If they are using MQTT it would get loaded from `rpi_power` instead of the setup of `mqtt`